### PR TITLE
Feature/cs732 19

### PR DESCRIPTION
**Pull request by MarkZhuVUW**
1. Add Playwright project at root folder & one example test
2. Add app `./Dockerfile` into `./docker-compose` file
3. Add Playwright integration tests into Github Actions `ci.yml`

To test running playwright:
```
cd ./scripts
sh ./start.sh
cd ..
npm run test:e2e
```
Will tidy up README once https://github.com/UOA-CS732-SE750-Students-2024/project-group-careless-crayfish/pull/5 is merged

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -49,14 +46,92 @@ jobs:
...
+
       - name: Enable backend service and test
+
         working-directory: "./backend"
         run: |
-          (npm run start&)
-          sleep 10
-          npm run test:e2e
\ No newline at end of file
+          echo "MONGO_SERVER=$MONGO_SERVER" >> .env
```
[the remark this answers is not in the document]
Comment (MarkZhuVUW) [reply .github/workflows/ci.yml:58]: config/creds are not supposed to be checked in. I have a ticket to move them onto circle ci

**Comment by MarkZhuVUW** [comment]
> image the `act` build doesn't seem working?
> 
> try `act -j chrome-e2e-test`?



> image the `act` build doesn't seem working?

try `act -j chrome-e2e-test`?

**Comment by MarkZhuVUW** [comment]
> image THe e2e test doesn't look working

yep good catch. 

Seems like I have not included instruction to install playwright and its browsers

try `npx playwright install --with-deps chromium`

included readme as well

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarkZhuVUW) [reply .github/workflows/ci.yml:58]: https://aucklanduni-team-cs732.atlassian.net/browse/CS732-24

File: backend/daos/mongoConnection.js [hunk 2]
```diff
@@ -1,19 +1,24 @@
+/* eslint-disable no-undef */
 const mongoose = require("mongoose");
-const Config = require("config");
 
 async function connect() {
-  const address = Config.get('application.mongo.address');
```
Comment (MarkZhuVUW) [inline backend/daos/mongoConnection.js:5]: I can't get the npm config library working with docker-compose either in CI env or running it locally.

So I have instead actually been loading these configs/credentials from a .env file under the backend folder...

**Comment by KevinHChen** [comment]
The code is not working, please take a look. As it's a demo project, I recommend to remove the configuration and set everything as `const string` inside the codes to simplify the logic. I recommend us to focus more on the business logic.

Yep, it's my fault to introduce the `config` package. Sorry about that. But it's not yet too late to fix the issue

image

**Review by KevinHChen — APPROVED** [review]
Looks good now

**Pull request merged**